e2e-tests/src/ban.rs

```diff
@@ -1,4 +1,4 @@
-use std::collections::HashSet;
+use std::collections::BTreeSet;
```

Comment:
Same problem with `Hash`, but for tests imo we can apply that ugly quick fix - let me know what do you think

Reply:
I guess we are safe, modulo problems with repeated validators, which actually should never occur...

Reply:
Rolled back